lib/pf/pki_provider/packetfence_pki.pm

```diff
@@ -99,28 +95,12 @@ Revoke the certificate for a user
 sub revoke {
     my ($self, $cn) = @_;
     my $logger = get_logger();
-    my %options = (
-        -where => {
-            id => "/pki/$cn",
-        }
-    );
-
-    my ($status, $iter) = pf::dal::key_value_storage->search(%options);
-    if (is_success($status)) {
-        my $key_value = $iter->next(undef);
-        if ($key_value) {
-            my $return = pf::api::unifiedapiclient->default_client->call("DELETE", "/api/v1/pki/cert/$key_value->{value}/$cn/1");
-        } else {
-            $logger->error("Unable to find certificate in the cache $cn");
-        }
-    } else {
-        $logger->error("Unable to revoke user certificate $cn");
-    }
 
-    ($status, my $count) = pf::dal::key_value_storage->remove_items(%options);
-    
-    if (is_error($status)) {
-        $logger->error("Unable to delete the key value '$cn'");
+    my $value = eval {
+        my $return = pf::api::unifiedapiclient->default_client->call("DELETE", "/api/v1/pki/cert/$self->{'profile'}/$cn/1");
+    };
+    if ($@) {
+        $logger->debug("Unable to revoke certificate associated with this cn $cn");
```

Comment:
Missing `return` after this line ?

Reply:
we don´t test the return value.
The code revoke the cert if it exist and do nothing if it doesn´t exist.